File: echo-guardian-production-foundation-v1.0/native/android/echo-guardian-core/src/main/java/org/echoguardian/export/SignedNativeExportManifest.kt
package org.echoguardian.export

data class SignedNativeExportManifest(
    val schemaVersion: String = "0.7",
    val platform: String = "android",
    val exportManifestHash: String,
    val signatureAlgorithm: String = "ed25519_persisted_contract",
    val signatureHex: String,
    val publicKeyHex: String,
    val userConfirmed: Boolean = true,
    val automaticExport: Boolean = false,
    val plainLanguageSummary: String = "Echo Guardian signed a user-confirmed native export manifest. No automatic export was used."
)

object SignedNativeExportPolicy {
    fun validate(userConfirmed: Boolean, automaticExport: Boolean, emergencyServicesUsed: Boolean): Boolean =
        userConfirmed && !automaticExport && !emergencyServicesUsed
}

File: echo-guardian-production-foundation-v1.0/native/ios/EchoGuardianKit/Sources/EchoGuardianKit/Export/SignedNativeExportManifest.swift
import Foundation

public struct SignedNativeExportManifest: Codable, Equatable {
    public let schemaVersion: String
    public let platform: String
    public let exportManifestHash: String
    public let signatureAlgorithm: String
    public let signatureHex: String
    public let publicKeyHex: String
    public let userConfirmed: Bool
    public let automaticExport: Bool
    public let plainLanguageSummary: String

    public init(exportManifestHash: String, signatureHex: String, publicKeyHex: String) {
        self.schemaVersion = "0.7"
        self.platform = "ios"
        self.exportManifestHash = exportManifestHash
        self.signatureAlgorithm = "ed25519_persisted_contract"
        self.signatureHex = signatureHex
        self.publicKeyHex = publicKeyHex
        self.userConfirmed = true
        self.automaticExport = false
        self.plainLanguageSummary = "Echo Guardian signed a user-confirmed native export manifest. No automatic export was used."
    }
}

public enum SignedNativeExportPolicy {
    public static func validate(userConfirmed: Bool, automaticExport: Bool, emergencyServicesUsed: Bool) -> Bool {
        userConfirmed == true && automaticExport == false && emergencyServicesUsed == false
    }
}
